packages/lib-classifier/src/components/Classifier/components/withKeyZoom/README.md

```diff
@@ -0,0 +1,33 @@
+# withKeyZoom
+
+A decorator that connects a component to the pan and zoom actions in the `SubjectViewer` store, which handles the image toolbar functions. The wrapped component is given an `onKeyDown` handler but focus management is the responsibility of the component author, not this <abbr title="Higher-Order Component">HOC</abbr>.
+
+## keyboard mappings
+
+<dl>
+  <dt>+</dt>
+  <dd>`SubjectViewer.zoomIn`</dd>
+  <dt>-</dt>
+  <dd>`SubjectViewer.zoomOut`</dd>
+  <dt>left arrow</dt>
+  <dd>`SubjectViewer.panLeft`</dd>
+  <dt>right arrow</dt>
+  <dd>`SubjectViewer.panRight`</dd>
+  <dt>up arrow</dt>
+  <dd>`SubjectViewer.panUp`</dd>
+  <dt>down arrow</dt>
+  <dd>`SubjectViewer.panDown`</dd>
+</dl>
+
+## Usage
+
+```
+  withKeyZoom(MyComponent)
+```
+or with another connected component
+```
+  @inject(storeMapper)
+  @withKeyZoom
+  @observer
+  class MyComponent { }
+```
```

Comment:
I think we need to clarify that this is the older mobx-react method of decorating and show and example with v6:

```
with another connected class component using mobx-react v5

  @inject(storeMapper)
  @withKeyZoom
  @observer
  class MyComponent { }


with another connected class or functional component using mobx-react v6 (preferred)

import { observer, MobXProviderContext } from 'mobx-react'
function useStores() {
  return React.useContext(MobXProviderContext)
}

class MyComponent { 
  render () {
    const stores = useStores()
   }
}

export withKeyZoom(observer(MyComponent))

function MyFunctionalComponent ()  {
   const stores = useStores()
   return ()
}

export withKeyZoom(observer(MyFunctionalComponent))

```

Feel free to tweak my suggestion here. I couldn't write this as an actual suggestion and get the formatting right in the github markdown.

Reply:
Maybe leave out inject and write it as
```js
@withKeyZoom
@observer
class MyComponent { }
```
or
```js
withKeyZoom(observer(MyComponent))
```
depending on how you like to write decorators. Do we have a style preference for decorating classes with the `@decorator` syntax?

We maybe don't need to show how to inject properties into a component, since that would be repeating the MobX documentation.

Reply:
Yeah, leaving out the example might be best since there are a few different ways it can be done and just add a note about referencing the correct mobx-react docs depending on the version with a link to it?

Reply:
I can update `withKeyZoom` to replace `inject` with context too.

Reply:
Sounds good. I'd like to start to move us over to the v6 way of doing things.

Reply:
I think all this needs to note is that the HOC should wrap `observer` when used with observed components. Something like that. `observer(withKeyZoom(MyComponent))` won't work.

There might already be a note about decorating observers in the mobx-react docs, in which case I'll link to it.